src/palette/qml/MuseScore/Palette/internal/TreePaletteHeader.qml

```diff
@@ -132,7 +132,17 @@ Item {
         acceptedButtons: Qt.RightButton
 
         onClicked: {
-            menuButton.toggleMenu(this, mouseX, mouseY)
+            contextMenuLoader.show(Qt.point(mouseX, mouseY))
+        }
+
+        ContextMenuLoader {
```

Comment:
I don't quite understand why the second menu is needed... Can't we just close the already open menu if the user clicks somewhere?

Reply:
The ContextMenuLoader displays the context menu when the header of the palette is right-clicked as opposed to when the three-dots button is clicked. The ContextMenuLoader positions itself at the passed coordinates and makes itself the parent of the menu. Thus the context menu appears under the coordinates where the user right-clicked. Also this makes the menu close when the user clicks on the header while the menu is displayed because the header is not the parent. So both issues in #21553 are easily solved with the ContextMenuLoader and just a few lines of code.